File: apps/api/src/lib/analytics.go
package lib

import (
	"context"
	"os"
	"strings"

	mp "github.com/mixpanel/mixpanel-go"
)

var mpClient *mp.ApiClient

func InitAnalytics() {
	// ITAR/CUI boundary: never egress analytics from the GovCloud partition,
	// even if a token is configured. Mixpanel ingests outside the boundary.
	if strings.HasPrefix(strings.ToLower(os.Getenv("AWS_REGION")), "us-gov-") {
		return
	}
	token := os.Getenv("MIXPANEL_TOKEN")
	if token != "" {
		mpClient = mp.NewApiClient(token)
	}
}

func Track(event string, distinctID string, props map[string]any) {
	if mpClient == nil {
		return
	}
	mpClient.Track(context.Background(), []*mp.Event{
		mpClient.NewEvent(event, distinctID, props),
	})
}

File: apps/web/src/lib/analytics.ts
import mixpanel from 'mixpanel-browser'

// ITAR/CUI boundary: never load analytics in the GovCloud build, regardless of
// token configuration. Mixpanel's ingestion endpoint is outside the boundary.
// The gov web bundle bakes a us-gov-* region, so gate on that (fail-safe: even
// if a token were baked in, IS_GOV forces analytics off).
const IS_GOV = (process.env.NEXT_PUBLIC_COGNITO_REGION || '').toLowerCase().startsWith('us-gov-')
const TOKEN = IS_GOV ? '' : (process.env.NEXT_PUBLIC_MIXPANEL_TOKEN || '')
let initialized = false

function init() {
  if (initialized || !TOKEN) return
  mixpanel.init(TOKEN, { track_pageview: false, persistence: 'localStorage' })
  initialized = true
}

export function track(event: string, props?: Record<string, any>) {
  if (!TOKEN) return
  init()
  mixpanel.track(event, props)
}

export function identify(id: string, traits?: Record<string, any>) {
  if (!TOKEN) return
  init()
  mixpanel.identify(id)
  if (traits) mixpanel.people.set(traits)
}

export function reset() {
  if (!TOKEN) return
  init()
  mixpanel.reset()
}
